processing/src/test/java/org/apache/druid/query/groupby/UnnestGroupByQueryRunnerTest.java

```diff
@@ -0,0 +1,833 @@
+    for (GroupByQueryConfig config : testConfigs()) {
+      final GroupByQueryRunnerFactory factory = makeQueryRunnerFactory(config, BUFFER_POOLS);
+      for (QueryRunner<ResultRow> runner : QueryRunnerTestHelper.makeUnnestQueryRunners(
+          factory,
+          QueryRunnerTestHelper.PLACEMENTISH_DIMENSION,
+          QueryRunnerTestHelper.PLACEMENTISH_DIMENSION_UNNEST,
+          null
+      )) {
+        for (boolean vectorize : ImmutableList.of(false)) {
+          final String testName = StringUtils.format("config=%s, runner=%s, vectorize=%s", config, runner, vectorize);
+
+          // Add vectorization tests for any indexes that support it.
+          if (!vectorize ||
+              (QueryRunnerTestHelper.isTestRunnerVectorizable(runner) &&
+               config.getDefaultStrategy().equals(GroupByStrategySelector.STRATEGY_V2))) {
+            constructors.add(new Object[]{testName, config, factory, runner, vectorize});
+          }
+        }
+      }
+    }
+
+    return constructors;
+  }
+
+  @BeforeClass
+  public static void setUpClass()
+  {
+    if (BUFFER_POOLS == null) {
+      BUFFER_POOLS = TestGroupByBuffers.createDefault();
+    }
+  }
+
+  @AfterClass
+  public static void tearDownClass()
+  {
+    BUFFER_POOLS.close();
+    BUFFER_POOLS = null;
+  }
+
+  private static ResultRow makeRow(final GroupByQuery query, final String timestamp, final Object... vals)
+  {
+    return GroupByQueryRunnerTestHelper.createExpectedRow(query, timestamp, vals);
+  }
+
+  private static ResultRow makeRow(final GroupByQuery query, final DateTime timestamp, final Object... vals)
+  {
+    return GroupByQueryRunnerTestHelper.createExpectedRow(query, timestamp, vals);
+  }
+
+  private static List<ResultRow> makeRows(
+      final GroupByQuery query,
+      final String[] columnNames,
+      final Object[]... values
+  )
+  {
+    return GroupByQueryRunnerTestHelper.createExpectedRows(query, columnNames, values);
+  }
+
+  @Test
+  public void testGroupBy()
+  {
+    GroupByQuery query = makeQueryBuilder()
+        .setDataSource(UnnestDataSource.create(
+            new TableDataSource(QueryRunnerTestHelper.DATA_SOURCE),
+            QueryRunnerTestHelper.PLACEMENTISH_DIMENSION,
+            QueryRunnerTestHelper.PLACEMENTISH_DIMENSION_UNNEST,
+            null
+        ))
+        .setQuerySegmentSpec(QueryRunnerTestHelper.FIRST_TO_THIRD)
+        .setDimensions(new DefaultDimensionSpec("quality", "alias"))
+        .setAggregatorSpecs(
+            QueryRunnerTestHelper.ROWS_COUNT,
+            new LongSumAggregatorFactory("idx", "index")
+        )
+        .setGranularity(QueryRunnerTestHelper.DAY_GRAN)
+        .build();
+
+    List<ResultRow> expectedResults = Arrays.asList(
+        makeRow(
+            query,
+            "2011-04-01",
+            "alias",
+            "automotive",
+            "rows",
+            2L,
+            "idx",
+            270L
+        ),
```

Comment:
style nit, I don't think these are actually wide enough to require each of the arguments being on independent lines.  The test would read a lot easier if each of the `makeRow()` calls was its own line.